### PR TITLE
Donal spy fixes

### PR DESCRIPTION
**Pull request by Christian-B**
Cherry picks from https://github.com/SpiNNakerManchester/sPyNNaker/pull/701 which have nothing to do with the graph mapper

@andrewgait please recheck the connection changes the rest look fine to me.

tested with https://github.com/SpiNNakerManchester/sPyNNaker8/pull/364

### CODE REVIEW

**Comment by dkfellows** [comment]
Failure: 
```
TypeError: _write_synaptic_matrix_and_master_population_table() takes 14 positional arguments but 15 were given
```

File: spynnaker/pyNN/models/neural_projections/connectors/fixed_probability_connector.py [hunk 1]
```diff
@@ -137,53 +137,41 @@ def create_synaptic_block(
...
         if synapse_info.postpop_is_view:
-            post_view_lo, post_view_hi = self._get_view_lo_hi(
-                synapse_info.post_population._indexes)
-
-        params.extend([post_view_lo, post_view_hi])
+            view_range = self._get_view_lo_hi(synapse_info.post_population)
+        params.extend(view_range)
 
-        params.extend([self.__allow_self_connections])
+        params.append(self.__allow_self_connections)
```
Comment (andrewgait) [inline spynnaker/pyNN/models/neural_projections/connectors/fixed_probability_connector.py]: Is there a reason why .extend has been replaced with .append here?

File: spynnaker/pyNN/models/neural_projections/connectors/fixed_probability_connector.py [hunk 2]
```diff
@@ -137,53 +137,41 @@ def create_synaptic_block(
...
 
         # If prob=1.0 has been specified, take care when scaling value to
         # ensure that it doesn't wrap round to zero as an unsigned long fract
-        params.extend([DataType.U032.encode_as_int(
-            DataType.U032.max if self._p_connect == 1.0 else self._p_connect)])
+        params.append(DataType.U032.encode_as_int(
+            DataType.U032.max if self._p_connect == 1.0 else self._p_connect))
```
Comment (andrewgait) [inline spynnaker/pyNN/models/neural_projections/connectors/fixed_probability_connector.py]: As above .extend -> .append?

**Comment by dkfellows** [comment]
Oh dear, none of this will work without the graph mapper changes.

File: spynnaker/pyNN/models/neural_projections/connectors/kernel_connector.py [hunk 3]
```diff
@@ -14,14 +14,14 @@
...
 import numpy
-from pyNN.random import RandomDistribution
-from .abstract_connector import AbstractConnector
-from spynnaker.pyNN.exceptions import SpynnakerException
 from spinn_utilities.overrides import overrides
 from data_specification.enums.data_type import DataType
 from spinn_front_end_common.utilities.constants import BYTES_PER_WORD
-from .abstract_generate_connector_on_machine \
-    import AbstractGenerateConnectorOnMachine, ConnectorIDs, PARAM_TYPE_KERNEL
+from pyNN.random import RandomDistribution
```
Comment (andrewgait) [inline spynnaker/pyNN/models/neural_projections/connectors/kernel_connector.py]: I thought it was standard procedure to put "third party" froms / imports before SpiNNaker ones?

**Comment by andrewgait** [comment]
Ah and I just got to where Donal was 5 minutes ago... yeah, the synaptic_manager changes shouldn't be here.

File: spynnaker/pyNN/models/neural_projections/connectors/fixed_probability_connector.py (hunk 1 above)
Comment (dkfellows) [reply spynnaker/pyNN/models/neural_projections/connectors/fixed_probability_connector.py]: Yes. We were making a list of one element to just add a single element to an ordinary list. Which is very silly.

File: spynnaker/pyNN/models/neural_projections/connectors/kernel_connector.py (hunk 3 above)
Comment (dkfellows) [reply spynnaker/pyNN/models/neural_projections/connectors/kernel_connector.py]: Fixed

**Comment by Christian-B** [comment]
main grtaph mapper is in this is dead

**Pull request closed without merging**